core/cog-gamepad-manette.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * gamepad.h
+ * Copyright (C) 2021 Igalia S.L.
+ *
+ * Distributed under terms of the MIT license.
+ */
+
+#include "cog-gamepad.h"
+
+#include <libmanette.h>
+#include <linux/input-event-codes.h>
+#include <wpe/wpe.h>
+
+typedef struct _Provider Provider;
+typedef struct _Gamepad  Gamepad;
+
+struct _Provider {
+    struct wpe_gamepad_provider *server;
+    ManetteMonitor              *monitor;
+
+    gulong connected;
+    gulong disconnected;
+};
+
+struct _Gamepad {
+    struct wpe_gamepad *server;
```

Comment:
The name “server” here is odd. There is no server, because the code uses libmanette in-process. This could be just `struct wpe_gamepad *gamepad`.

Reply:
I renamed `gamepad` and `provider` from WebKit as `wk_gamepad` and `wk_provider`. Otherwise there're symbols collisions.